### DISCUSSION TITLE
Selecting Intellisense suggestion insertion question

### DISCUSSION THREAD

**Discussion by jalanh11640** [microsoft/vscode-cpptools#11783] [Q&A] [answered]
First of all, my apologies if this isn't an extension specific question.

If I have a line like this..

Entry& entry = someEntry;

If I want to scope Entry, I'll put the cursor to the left of Entry and start typing.  Intellisense will pop up with suggestions.  If I select the class name I'm interested in, Entry will be replaced with my selection.  Is there a setting to tell the intellisense to insert the selection rather than replace what is ahead of it?

Hopefully, this question made sense.
Thanks,
Jeff

**Comment by bobbrow** [comment, marked as answer]
We enabled this behavior in 1.19 which is still in pre-release.  See https://github.com/microsoft/vscode-cpptools/issues/10613.

If you install the pre-release version of the extension, you can add `"[cpp]": { "editor.suggest.insertMode": "insert" }` to your settings to enable insert mode.

**Reply by jalanh11640** [reply]
Awesome, thank you!
